### ISSUE TITLE
Doesn't build on OS X

### ISSUE THREAD

**Issue by developerworks** [goertzenator/rust.mk#1]
What version of rust you use, I get the following errors by `make` in `tests/rustmk_demo` directory:

```
➜  rustmk_demo git:(master) make
 DEP    find_crate
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
 DEPEND find_crate.d
 ERLC   find_crate.erl
 APP    find_crate
 DEPEND rustmk_demo.d
 ERLC   complex1.erl ruster_unsafe_demo.erl
 APP    rustmk_demo
 CARGO  erl_comm
 CARGO  ruster_unsafe_demo
error: linking with `cc` failed: exit code: 1
note: "cc" "-m64" "-L" "/Users/hezhiqiang/.multirust/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/ruster_unsafe_demo/target/release/ruster_unsafe_demo.0.o" "-o" "/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/ruster_unsafe_demo/target/release/libruster_unsafe_demo.dylib" "/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/ruster_unsafe_demo/target/release/ruster_unsafe_demo.metadata.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/ruster_unsafe_demo/target/release" "-L" "/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/ruster_unsafe_demo/target/release/deps" "-L" "/Users/hezhiqiang/.multirust/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/libruster_unsafe-badf32d60ecc0604.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/liblibc-38919d24e617a235.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/libstd-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/libcollections-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/librustc_unicode-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/librand-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/liballoc-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/liballoc_system-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/liblibc-d16b8f0e.rlib" "-Wl,-force_load,/var/folders/m3/vn1yyrfx7l36cq6p8msc6wt80000gn/T/rustc.baCXp1mKElZd/libcore-d16b8f0e.rlib" "-l" "System" "-l" "pthread" "-l" "c" "-l" "m" "-dynamiclib" "-Wl,-dylib" "-l" "compiler-rt"
note: Undefined symbols for architecture x86_64:
  "_enif_get_ulong", referenced from:
      ruster_unsafe::enif_get_uint64::h84c88c1e02aa4f6b in libruster_unsafe-badf32d60ecc0604.rlib(ruster_unsafe-badf32d60ecc0604.0.o)
  "_enif_make_atom", referenced from:
      ruster_unsafe_demo::load::hd7ee4ce1f3930938 in ruster_unsafe_demo.0.o
  "_enif_get_long", referenced from:
      ruster_unsafe::enif_get_int64::hfa5611b090fa7114 in libruster_unsafe-badf32d60ecc0604.rlib(ruster_unsafe-badf32d60ecc0604.0.o)
  "_enif_get_tuple", referenced from:
      ruster_unsafe_demo::tuple_add::hbffd5e5909b3abd6 in ruster_unsafe_demo.0.o
  "_enif_get_int", referenced from:
      ruster_unsafe_demo::native_add::h7ad592db04cbbd9b in ruster_unsafe_demo.0.o
      ruster_unsafe_demo::tuple_add::hbffd5e5909b3abd6 in ruster_unsafe_demo.0.o
  "_enif_make_int", referenced from:
      ruster_unsafe_demo::native_add::h7ad592db04cbbd9b in ruster_unsafe_demo.0.o
      ruster_unsafe_demo::tuple_add::hbffd5e5909b3abd6 in ruster_unsafe_demo.0.o
  "_enif_make_ulong", referenced from:
      ruster_unsafe::enif_make_uint64::h3d5f5def6c89f123 in libruster_unsafe-badf32d60ecc0604.rlib(ruster_unsafe-badf32d60ecc0604.0.o)
  "_enif_make_long", referenced from:
      ruster_unsafe::enif_make_int64::hf48b56b9d1478290 in libruster_unsafe-badf32d60ecc0604.rlib(ruster_unsafe-badf32d60ecc0604.0.o)
  "_enif_make_badarg", referenced from:
      ruster_unsafe_demo::native_add::h7ad592db04cbbd9b in ruster_unsafe_demo.0.o
      ruster_unsafe_demo::tuple_add::hbffd5e5909b3abd6 in ruster_unsafe_demo.0.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

error: aborting due to previous error
error: Could not compile `ruster_unsafe_demo`.

To learn more, run the command again with --verbose.
make: *** [/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/ruster_unsafe_demo] Error 101
```

**Comment by goertzenator** [comment]
This is an OS X build problem for ruster unsafe, ref https://github.com/goertzenator/ruster_unsafe/issues/3 .  I saw you work through this at https://github.com/hansihe/Rustler/issues/16, but I'd like to solve this in rust.mk so the Rust source doesn't have to be altered.

Can you run make like this and show me the results?

``` bash
RUSTFLAGS=--codegen link-args='-flat_namespace -undefined suppress' make tests
```

If it works I'll embed the `RUSTFLAGS` into `rust.mk`.  You may run into `find_crate` problems too..I'd also like fix that for OS X.

**Comment by developerworks** [comment]
The output is:

```
➜  rustmk_demo git:(master) ✗ RUSTFLAGS="--codegen link-args='-flat_namespace -undefined suppress'" make tests          
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
make[1]: Nothing to be done for `test-dir'.
 CARGO  erl_comm
error: Unrecognized option: 'u'.
error: Could not compile `byteorder`.

To learn more, run the command again with --verbose.
make: *** [/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/erl_comm] Error 101
```

**Comment by goertzenator** [comment]
Thanks.  How about with erlang.mk verbose:

```
RUSTFLAGS="--codegen link-args='-flat_namespace -undefined suppress'" make V=1 tests 
```

**Comment by developerworks** [comment]
```
➜  rustmk_demo git:(master) ✗ RUSTFLAGS="--codegen link-args='-flat_namespace -undefined suppress'" make V=1 tests
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
for dep in  ; do \
        /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_APP=1 || exit $?; \
    done
rm -f /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log
mkdir -p /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk
for dep in /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/deps/find_crate ; do \
        if grep -qs ^$dep$ /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; then \
            :; \
        else \
            echo $dep >> /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; \
            if [ -f $dep/GNUmakefile ] || [ -f $dep/makefile ] || [ -f $dep/Makefile ]; then \
                /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_DEP=1 || exit $?; \
            else \
                echo "Error: No Makefile to build dependency $dep."; \
                exit 2; \
            fi \
        fi \
    done
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
for dep in  ; do \
        /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_APP=1 || exit $?; \
    done
mkdir -p /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk
for dep in  ; do \
        if grep -qs ^$dep$ /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; then \
            :; \
        else \
            echo $dep >> /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; \
            if [ -f $dep/GNUmakefile ] || [ -f $dep/makefile ] || [ -f $dep/Makefile ]; then \
                /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_DEP=1 || exit $?; \
            else \
                echo "Error: No Makefile to build dependency $dep."; \
                exit 2; \
            fi \
        fi \
    done
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory app-build
:
:
for dep in  ; do /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_DEP=1; done
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory app-build test-dir ERLC_OPTS="+debug_info +warn_export_vars +warn_shadow_vars +warn_obsolete_guard -DTEST=1"
:
make[1]: Nothing to be done for `test-dir'.
cd /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/erl_comm && cargo build -q --release
error: Unrecognized option: 'u'.
error: Could not compile `byteorder`.

To learn more, run the command again with --verbose.
make: *** [/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/erl_comm] Error 101
```

**Comment by goertzenator** [comment]
Thanks again.  RUSTFLAGS isn't going to work here.  Please pull the [osx test branch](https://github.com/goertzenator/rust.mk/tree/osx) and try that one with just...

``` bash
make V=1 tests
```

This branch also tries to rename the .dylib to a .so after it's copied to the `priv` dir.  Crossing my fingers!

**Comment by developerworks** [comment]
```
➜  rustmk_demo git:(osx) ✗ make V=1 tests
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
for dep in  ; do \
        /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_APP=1 || exit $?; \
    done
rm -f /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log
mkdir -p /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk
for dep in /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/deps/find_crate ; do \
        if grep -qs ^$dep$ /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; then \
            :; \
        else \
            echo $dep >> /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; \
            if [ -f $dep/GNUmakefile ] || [ -f $dep/makefile ] || [ -f $dep/Makefile ]; then \
                /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_DEP=1 || exit $?; \
            else \
                echo "Error: No Makefile to build dependency $dep."; \
                exit 2; \
            fi \
        fi \
    done
for dep in  ; do \
        mkdir -p $dep/ebin; \
    done
for dep in  ; do \
        /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_APP=1 || exit $?; \
    done
mkdir -p /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk
for dep in  ; do \
        if grep -qs ^$dep$ /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; then \
            :; \
        else \
            echo $dep >> /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/.erlang.mk/deps.log; \
            if [ -f $dep/GNUmakefile ] || [ -f $dep/makefile ] || [ -f $dep/Makefile ]; then \
                /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_DEP=1 || exit $?; \
            else \
                echo "Error: No Makefile to build dependency $dep."; \
                exit 2; \
            fi \
        fi \
    done
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory app-build
:
:
for dep in  ; do /Applications/Xcode.app/Contents/Developer/usr/bin/make -C $dep IS_DEP=1; done
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory app-build test-dir ERLC_OPTS="+debug_info +warn_export_vars +warn_shadow_vars +warn_obsolete_guard -DTEST=1"
:
make[1]: Nothing to be done for `test-dir'.
cd /Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/erl_comm && cargo build -q --release -- --codegen link-args='-flat_namespace -undefined suppress'
error: Invalid arguments.

Usage:
    cargo build [options]
make: *** [/Users/hezhiqiang/RustProjects/rust.mk/tests/rustmk_demo/crates/erl_comm] Error 1
```

**Comment by goertzenator** [comment]
I updated the `osx` branch.  Please give it another try.  I think we're almost there.

**Comment by goertzenator** [comment]
Fixed by 1bf843b8ff91b7abee5701f4b5148a0b9fe184a1.
